packages/baseline-browser-mapping/src/scripts/baseline-browser-versions.ts

```diff
@@ -0,0 +1,387 @@
+    );
+  }
+
+  const compatibleFeatures = getCompatibleFeaturesByDate(date);
+  const minimumVersions = getMinimumVersionsFromFeatures(compatibleFeatures);
+
+  if (!listAllCompatibleVersions) {
+    return minimumVersions;
+  } else {
+    return [...minimumVersions, ...getSubsequentVersions(minimumVersions)].sort(
+      (a, b) => {
+        if (a.browser < b.browser) {
+          return -1;
+        } else if (a.browser > b.browser) {
+          return 1;
+        } else {
+          return compareVersions(a.version, b.version);
+        }
+      },
+    );
+  }
+};
+
+const getDownstreamBrowsers = (
+  inputArray: BrowserVersion[] = [],
+  listAllCompatibleVersions: boolean = true,
+): BrowserVersion[] => {
+  let minimumChromeVersion: string | undefined = undefined;
+  if (inputArray && inputArray.length > 0) {
+    minimumChromeVersion = inputArray
+      .filter((browser: BrowserVersion) => browser.browser === "chrome")
+      .sort((a: BrowserVersion, b: BrowserVersion) => {
+        return compareVersions(a.version, b.version);
+      })[0]?.version;
+  }
+
+  if (!minimumChromeVersion) {
+    throw new Error(
+      "There are no browser versions compatible with Baseline before Chrome",
+    );
+  }
+
+  let downstreamArray: BrowserVersion[] = new Array();
+
+  downstreamBrowserData.forEach(([browserName, browserData]) => {
+    if (!browserData.releases) return;
+    let sortedAndFilteredVersions = Object.entries(browserData.releases)
+      .filter(([, versionData]) => {
+        if (!versionData.engine) {
+          return false;
+        }
+        if (versionData.engine != "Blink") {
+          return false;
+        }
+        if (
+          versionData.engine_version &&
+          parseInt(versionData.engine_version) < parseInt(minimumChromeVersion!)
+        ) {
+          return false;
+        }
+        return true;
+      })
+      .sort((a, b) => {
+        return compareVersions(a[0], b[0]);
+      });
+
+    for (let i = 0; i < sortedAndFilteredVersions.length; i++) {
+      const versionEntry = sortedAndFilteredVersions[i];
+      if (versionEntry) {
+        const [versionNumber, versionData] = versionEntry;
+        let outputObject: BrowserVersion = {
+          browser: browserName,
+          version: versionNumber,
+          release_date: versionData.release_date ?? "unknown",
+        };
+
+        if (versionData.engine && versionData.engine_version) {
+          outputObject.engine = versionData.engine;
+          outputObject.engine_version = versionData.engine_version;
+        }
+
+        downstreamArray.push(outputObject);
+
+        if (!listAllCompatibleVersions) {
+          break;
+        }
+      }
+    }
+  });
+  let outputArray = [...inputArray, ...downstreamArray];
+  return outputArray;
+};
+
+type Options = {
+  listAllCompatibleVersions?: boolean;
+  includeDownstreamBrowsers?: boolean;
+  widelyAvailableOnDate?: string | number;
+  targetYear?: number;
+};
```

Comment:
It could be useful to add descriptions for each of these options. Here's something I prototyped, as an example:

```suggestion
type Options = {
  listAllCompatibleVersions?: boolean;
  /**
   * Whether to include browsers that use the same engines as a core Baseline browser.
   */
  includeDownstreamBrowsers?: boolean;
  widelyAvailableOnDate?: string | number;
  targetYear?: number;
};
```

IDEs will show this description as part of the auto-suggest feature:

<img width="692" alt="image" src="https://github.com/user-attachments/assets/33dc681d-39bc-4031-9dc7-16cecaca76a6" />